riscv-sbi.adoc

```diff
@@ -1674,6 +1675,250 @@ The possible error codes returned in `sbiret.error` are shown in the
+struct sbiret sbi_measurement_extend(unsigned long measurement_addr,
+                                     unsigned long measurement_index,
+                                     unsigned long hash_algorithm)
+----
+
+Extend the platform TCB measurement with one additional measurement.
+
+This function typically allows the caller to add a software component to the
+list of the TCB layers measurements at any point of the platform lifecycle.
+By supporting this function, the SBI implementation implicitly becomes a
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[Dynamic Root of Trust for Measurement (DRTM)].
+
+Measurements added through this SBI function should be managed separately from
+the other TCB measurements. The SBI implementation can choose to maintain them
+through multiple measurement registers, or extend a single one for every
+`sbi_attestation_extend_measurement` call. In the former case, the caller can
+select a measurement index through the `measurement_index` parameter to extend
+one particular measurement register. When supporting multiple measurement
+registers, SBI implementations should map the supported indexes to the
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[TCG specified DTRM PCR indexes].
+
+All measurements added through this SBI function must be included in the TCB
+attestation evidence generated by the <<get-evidence-function>> function.
+
+The measurement at the `measurement_addr` physical address must be generated
+by a hash function that uses the algorithm specified by `hash_algorithm`.
+`hash_algorithm` must be one of the supported hash algorithms, as reported by
+the `HASH_ALGORITHMS` field from the <<get-capabilities-function>> returned
+structure (`ATST_CAPS`). See
+<<table_attestation_and_measurement_capabilities_format>> for more details.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_measurement_extend_errors>> below.
+
+[#table_measurement_extend_errors]
+.Measurement Extend Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The measurement was succesfully extended.
+| SBI_ERR_FAILED        | The measurement could not be extended.
+| SBI_INVALID_PARAM     | The `measurement_index` is invalid.
+| SBI_ERR_NOT_SUPPORTED | The function is not supported as the measurement
+                          can not be extended.
+|===
+
+[#get-capabilities-function,reftext="Get Capabilities"]
+=== Function: Get Attestation and Measurement Capabilities (FID #2)
+
+[source, C]
+----
+struct sbiret sbi_attestation_get_capabilities(unsigned long caps_addr)
+----
+
+Get the SBI implementation attestation and measurement capabilities.
+
+The reported capabilities allow attestation and measurement implementations to
+build platform agnostic frameworks, by dynamically adapting the SBI calls to the
```

Comment:
Consider terminating the sentence at "....platform agnostic frameworks.". The ", by dynamically seems a little redundant, and slightly confusing".